lib/server.js

```diff
@@ -112,6 +121,15 @@ module.exports = function createServer(config, log) {
       }
       email = normalizedEmail(email)
 
+      // Check blocklist before retrieving records, save a lookup
```

Comment:
While saving lookups is usually a good idea, this will bypass the logic that checks for `allowedIPs` and `allowedEmails`, as well as bypassing some metrics logging.

I think we'll need to move this somewhere inside the `fetchRecords` handler, and/or move the `allowedIPs` logic outside of that handler; the private repo has a bit of a refactor of this code that may help here, see https://github.com/mozilla/fxa-customs-server/issues/126

My general philosophy for this codebase, is to optimize for returning `block: false` as quickly as possible, but it's OK to do extra work for requests that we're going to block.  So doing an unnecessary `fetchRecords` for IPs that are on a blocklist, is OK by me.